### DISCUSSION TITLE
Redeploy wormhole wrapped Dai with 6 decimals

### DISCUSSION THREAD

**Discussion by michaelhly** [wormhole-foundation/wormhole#205] [Developer Q&A] [answered]
Currently, SPL USDC and USDT have 6 decimals. Dai has 18 decimals on Ethereum, and it must be shrunken down to 9 on Solana. 

Can we further just shrink the number of decimals down to 6? This makes it easier for dapps to integrate wormhole dai with the existing SPL USDC and USDT tokens.

**Comment by macalinao** [comment]
:+1: this would be huge!

**Comment by leoluk** [comment, marked as answer]
Wrapped tokens are deterministically generated by Wormhole:

https://github.com/certusone/wormhole/blob/3aee599adac50355bf3fac35a8aa0898a70da240/ethereum/contracts/Wormhole.sol#L259

There's no way to redeploy them or dynamically adjust the decimals. Is the SPL ecosystem generally using 6 decimal tokens, or are USDC and USDT a special case?

**Reply by michaelhly** [reply]
Yes. Serum, Raydium, Cope all only have 6 decimals:

https://explorer.solana.com/address/4k3Dyjzvzp8eMZWUXbBCjEvwSkkk59S5iCNLY3QrkX6R
https://explorer.solana.com/address/SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt
https://explorer.solana.com/address/8HGyAAB1yoM1ttS7pXjHMa3dukTFGQggnFFH3hJZgzQh

@leoluk, I think you guys should reconsider standardizing all wormhole tokens to 6 decimals. Having decimal mismatches creates unnecessary burden on Dapp developers and opens a wide surface area for bugs (especially decimal precision-related bugs).

**Reply by leoluk** [reply]
Yeah, that'd be the only option - have six decimals on _all_ wrapped tokens. We'll consider it - thanks for bringing it up!
